pulsar-metadata/src/main/java/org/apache/pulsar/metadata/bookkeeper/PulsarLedgerManager.java

```diff
@@ -106,14 +120,21 @@ public CompletableFuture<Versioned<LedgerMetadata>> createLedgerMetadata(long le
             return FutureUtil.failedFuture(new BKException.BKMetadataSerializationException(ioe));
         }
 
+        CompletableFuture<Versioned<LedgerMetadata>> promise = new CompletableFuture<>();
+
         CompletableFuture<Versioned<LedgerMetadata>> future = store.put(getLedgerPath(ledgerId), data, Optional.of(-1L))
-                .thenApply(stat -> new Versioned(metadata, new LongVersion(stat.getVersion())));
+                .thenApply(stat -> {
```

Comment:
Nit:
We can replace this with thenAccept
or the both of the two calls (apply and exceptionally) with whenComplete

Reply:
+1 for simplifying the code.

Reply:
fixed